internal/ethapi/api.go

```diff
@@ -1257,17 +1265,27 @@ func newRPCTransaction(tx *types.Transaction, blockHash common.Hash, blockNumber
 		result.BlockNumber = (*hexutil.Big)(new(big.Int).SetUint64(blockNumber))
 		result.TransactionIndex = (*hexutil.Uint64)(&index)
 	}
-	if tx.Type() == types.AccessListTxType {
+	switch tx.Type() {
+	case types.AccessListTxType:
+		al := tx.AccessList()
+		result.Accesses = &al
+		result.ChainID = (*hexutil.Big)(tx.ChainId())
+	case types.DynamicFeeTxType:
 		al := tx.AccessList()
 		result.Accesses = &al
 		result.ChainID = (*hexutil.Big)(tx.ChainId())
+		result.FeeCap = (*hexutil.Big)(tx.FeeCap())
+		result.Tip = (*hexutil.Big)(tx.Tip())
+		// price = min(tip, feeCap - baseFee) + baseFee
+		price := math.BigMin(new(big.Int).Add(tx.Tip(), baseFee), tx.FeeCap())
+		result.GasPrice = (*hexutil.Big)(price)
 	}
 	return result
 }
 
 // newRPCPendingTransaction returns a pending transaction that will serialize to the RPC representation
 func newRPCPendingTransaction(tx *types.Transaction) *RPCTransaction {
-	return newRPCTransaction(tx, common.Hash{}, 0, 0)
+	return newRPCTransaction(tx, common.Hash{}, 0, 0, nil)
```

Comment:
Is it possible to call this method with a 1559 tx?

Reply:
Yes it is, this actually points to a bigger issue, I should only be computing the effective gas price on _mined_ txs, for pending txs it should be `nil` since we don't know what the price will be.

I'll rework the code to account for that, and rebase this PR against `eip1559-misc` and break it up into two smaller PRs (one for the tx changes and one for adding baseFee to block responses).